src/renderer/components/Keybindings/index.tsx

```diff
@@ -0,0 +1,171 @@
+import React, { useContext } from 'react';
+import Styles from './styles.module.scss';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faInfoCircle } from '@fortawesome/free-solid-svg-icons';
+import { Action, useKeybindings } from 'renderer/hooks/useKeybindings';
+import { useHotkeyRecorder } from 'renderer/hooks/useHotkeysRecorder';
+import Modal from '../Modal';
+import { ChildWindowContext } from '../ChildWindow';
+
+export default function Keybindings() {
+  const { keybindings, setKeybinding } = useKeybindings();
+
+  const [recordingAction, setRecordingAction] = React.useState<Action | null>();
+
+  const [editModalOpen, setEditModalOpen] = React.useState(false);
+
+  const onSave = (keybinding: string) => {
+    if (!recordingAction) return;
+    setKeybinding(recordingAction, keybinding);
+    setEditModalOpen(false);
+  };
+
+  if (!keybindings) return null;
+
+  return (
+    <React.Fragment>
+      <Modal
+        title="Edit Keybinding"
+        open={editModalOpen}
+        onClose={() => setEditModalOpen(false)}
+      >
+        <EditKeybinding
+          onCancel={() => setEditModalOpen(false)}
+          onSave={onSave}
+        />
+      </Modal>
+      <div className={Styles.container}>
+        <main className={Styles.main}>
+          <div className={Styles.main__title}>Keybinds</div>
+          <div className={Styles.callout}>
+            <FontAwesomeIcon
+              icon={faInfoCircle}
+              className={Styles.callout__icon}
+            />
+            <p>All keybindings are disabled in this screen.</p>
+          </div>
+          <div className={Styles['keybinds-list']}>
+            {Object.entries(keybindings).map(([name, keybinding]) => {
+              return (
+                <div
+                  className={Styles['keybinds-list__item']}
+                  key={name}
+                  onClick={() => {
+                    setRecordingAction(name as Action);
+                    setEditModalOpen(true);
+                  }}
+                >
+                  <div>{name}</div>
+                  <div>
+                    <span className={Styles['keybinds-list__kbd']}>
+                      {keybinding.all}
+                    </span>
+                  </div>
+                </div>
+              );
+            })}
+            {/* <div className={Styles['keybinds-list__item']}>
+            <div>Run Query</div>
+            <div>
+              <span className={Styles['keybinds-list__kbd']}>
+                {process.platform === 'darwin' ? '⌘' : 'Ctrl'} + Enter
+              </span>
+            </div>
+          </div> */}
+          </div>
+        </main>
+      </div>
+    </React.Fragment>
+  );
+}
+
+type EditKeybindingProps = {
+  onSave: (keybinding: string) => void;
+  onCancel: () => void;
+};
+
+function EditKeybinding({ onSave, onCancel }: EditKeybindingProps) {
```

Comment:
You can use TextInput for handle the key recording instead of using hook and listen the document.